core/src/main/java/org/elasticsearch/search/DefaultSearchContext.java

```diff
@@ -168,7 +174,9 @@
         this.queryResult = new QuerySearchResult(id, shardTarget);
         this.fetchResult = new FetchSearchResult(id, shardTarget);
         this.indexShard = indexShard;
+        this.searchExecutor = indexShard.getThreadPool().executor(ThreadPool.Names.SEARCH);
```

Comment:
what I mean is instead of having the `searchExecutor` member can't we do this in `QueryPhase.execute` instead so we don't add more than necessary to the search context. it's already huge and we can access indexShard?